app/src/main/java/com/katahiromz/krakra_ja_jp/MyWebChromeClient.kt

```diff
@@ -32,26 +44,129 @@ class MyWebChromeClient(activity: MainActivity) : WebChromeClient() {
-        return super.onJsAlert(view, url, message, result)
+        val title = activity.getString(R.string.app_name)
+        MaterialDialog(activity).show {
+            title(text = title)
+            message(text = message)
+            positiveButton(text = "OK") { }
+        }
+        return true
     }
 
-    override fun onJsPrompt(
+    override fun onJsConfirm(
         view: WebView?,
         url: String?,
         message: String?,
-        defaultValue: String?,
-        result: JsPromptResult?
+        result: JsResult?
     ): Boolean {
-        return super.onJsPrompt(view, url, message, defaultValue, result)
+        val title = activity.getString(R.string.app_name)
+        MaterialDialog(activity).show {
+            title(text = title)
+            message(text = message)
+            positiveButton(text = "OK") {
+                result?.confirm()
+            }
+            negativeButton(text = "Cancel") {
+                result?.cancel()
+            }
+        }
+        return true
     }
 
-    override fun onJsConfirm(
+    @SuppressLint("CheckResult")
+    override fun onJsPrompt(
         view: WebView?,
         url: String?,
         message: String?,
-        result: JsResult?
+        defaultValue: String?,
+        result: JsPromptResult?
     ): Boolean {
-        return super.onJsConfirm(view, url, message, result)
+        val title = activity.getString(R.string.app_name)
+        if (isSelectMessageDialog(message)) {
+            // メッセージ選択ダイアログを表示
+            showSelectMessageDialog(title = title, message = message, result = result)
+            return true
+        }
+        var inputtedText: String? = null
+        MaterialDialog(activity).show {
+            title(text = title)
+            message(text = message)
+            input(hint = activity.getString(R.string.prompt_hint)) { _, text ->
+                inputtedText = text.toString()
+            }
+            positiveButton(text = "OK") {
+                result?.confirm(inputtedText ?: "")
+            }
+            negativeButton(text = "Cancel") {
+                result?.cancel()
+            }
+        }
+        return true
+    }
+
+    private fun isSelectMessageDialog(message: String?): Boolean = message == "メッセージ文字列を入力して下さい。"
+
+    private fun showSelectMessageDialog(
+        title: String,
+        message: String?,
+        result: JsPromptResult?
+    ) {
+        val dialog = MaterialDialog(activity).show {
+            title(text = title)
+            message(text = message)
+            customView(R.layout.message_select_dialog, scrollable = true, horizontalPadding = true)
+            positiveButton(text = "OK") {
+                val editText = getCustomView().findViewById<EditText>(R.id.message_edit)
+                result?.confirm(editText.text.toString())
+            }
+            negativeButton(text = "Cancel") {
+                result?.cancel()
+            }
+            lifecycleOwner(activity)
```

Comment:
このライフサイクルは何のためか説明できますか？

Reply:
lifecycleOwner の説明を見ると下記のようにありますので、ご参照ください。

```
Attach the dialog to a lifecycle and dismiss it when the lifecycle is destroyed. Uses the given owner lifecycle if provided, else falls back to the Context of the dialog window if it can.
Params:
owner --Optional lifecycle owner, if its null use windowContext.
```

要はダイアログインスタンスのメモリ管理を、指定したライフサイクルオーナーと同期するための宣言であり、今回は activity をライフサイクルオーナーとして指定しているため、activity が破棄されるのに合わせてダイアログインスタンスのメモリも破棄されることが保証されます。

ここ以外のダイアログの処理にもライフサイクルの指定を追加いたしました。